tiledb/sm/stats/stats.h

```diff
@@ -90,8 +90,14 @@ class Stats {
   /** Enable or disable statistics gathering. */
   void set_enabled(bool enabled);
 
-  /** Dumps the stats for this instance and all children in ASCII format. */
-  std::string dump() const;
+  /**
+   * Dumps the stats for this instance as a JSON dictionary of
+   * timers and stats.
+   *
+   * @param indent_size The number of spaces in an indentation.
+   * @param num_indents The number of leading indentations.
+   */
+  std::string dump(uint64_t indent_size, uint64_t num_indents) const;
```

Comment:
Should we choose some reasonable defaults here for `indent_size` and `num_indents`?